feature-cloud-backup-api/src/main/java/io/novafoundation/nova/feature_cloud_backup_api/domain/CloudBackupService.kt

```diff
@@ -69,9 +69,9 @@ suspend fun CloudBackupService.fetchAndDecryptExistingBackup(password: String):
  * @throws InvalidBackupPasswordError
  */
 suspend fun CloudBackupService.fetchAndDecryptExistingBackupWithSavedPassword(): Result<CloudBackup> {
-    return session.getSavedPassword().flatMap { savedPassword ->
-        fetchBackup().flatMap { encryptedBackup ->
-            encryptedBackup.decrypt(savedPassword)
+    return fetchBackup()
+        .flatMap { encryptedBackup ->
+            val password = session.getSavedPassword().getOrThrow()
+            encryptedBackup.decrypt(password)
```

Comment:
Iirc `getOrThrow` in `flatMap` wont catch exeptions so it will be propagated beyond Result
What is the reason for such a change? Old code looks correct

Reply:
I fixed it to get an error "BackupNotFound" before "PasswordNotSet", since it's important for case when we enable backup sync by switcher